packages/sit-onyx/src/components/OnyxButton/OnyxButton.vue

```diff
@@ -0,0 +1,157 @@
+
+    &:disabled {
+      --color: var(--onyx-color-text-icons-primary-soft);
+    }
+
+    &.onyx-button--default {
+      --background-color: var(--onyx-color-base-primary-500);
+      --background-hover-color: var(--onyx-color-base-primary-400);
+      --color: var(--onyx-color-text-icons-neutral-inverted);
+      --border-color: var(--onyx-color-base-primary-500);
+
+      &:disabled {
+        --background-color: var(--onyx-color-base-primary-200);
+        --border-color: var(--onyx-color-base-primary-200);
+        --color: var(--onyx-color-text-icons-neutral-inverted);
+      }
+    }
+
+    &.onyx-button--outline {
+      --border-color: var(--onyx-color-base-primary-500);
+
+      &:disabled {
+        --border-color: var(--onyx-color-base-primary-200);
+      }
+    }
+  }
+
+  &--secondary {
+    --background-hover-color: var(--onyx-color-base-neutral-200);
+    --focus-outline-color: var(--onyx-color-base-neutral-300);
+    --color: var(--onyx-color-text-icons-neutral-intense);
+
+    &:disabled {
+      --color: var(--onyx-color-text-icons-neutral-soft);
+    }
+
+    &.onyx-button--default {
+      --background-color: var(--onyx-color-base-background-blank);
+      --background-hover-color: var(--onyx-color-base-neutral-200);
+      --border-color: var(--onyx-color-base-neutral-400);
+
+      &:disabled {
+        --background-color: var(--onyx-color-base-background-blank);
+        --border-color: var(--onyx-color-base-neutral-200);
+      }
+    }
+
+    &.onyx-button--outline {
+      --border-color: var(--onyx-color-base-neutral-400);
+
+      &:disabled {
+        --border-color: var(--onyx-color-base-neutral-200);
+      }
+    }
+  }
+
+  &--danger {
+    --background-hover-color: var(--onyx-color-base-danger-200);
+    --focus-outline-color: var(--onyx-color-base-danger-300);
+    --color: var(--onyx-color-text-icons-danger-intense);
+
+    &:disabled {
+      --color: var(--onyx-color-text-icons-danger-medium);
+    }
+
+    &.onyx-button--default {
+      --background-color: var(--onyx-color-base-danger-200);
+      --background-hover-color: var(--onyx-color-base-danger-100);
+      --border-color: var(--onyx-color-base-danger-500);
+
+      &:disabled {
+        --background-color: var(--onyx-color-base-danger-100);
+        --border-color: var(--onyx-color-base-danger-200);
+      }
+    }
+
+    &.onyx-button--outline {
+      --border-color: var(--onyx-color-base-danger-500);
+
+      &:disabled {
+        --border-color: var(--onyx-color-base-danger-200);
+      }
+    }
+  }
+
+  background-color: var(--background-color, transparent);
+  border: var(--onyx-1px-in-rem) solid var(--border-color, transparent);
+
+  &:hover {
+    background: var(--background-hover-color, var(--onyx-color-base-primary-100));
+  }
+
+  &:focus-visible {
+    outline: 0.25rem solid var(--focus-outline-color, var(--onyx-color-base-primary-200));
+  }
+
+  &:disabled {
+    cursor: none;
+    pointer-events: none;
```

Comment:
```suggestion
```
a disabled element doesn't trigger events so we don't need this

Reply:
Actually we do have some side effects. When you hover over the disabled button the cursor disappears and the color changes as it's hovered